shell/src/main/java/alluxio/cli/Check.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.cli;
+
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+
+import com.google.common.base.Preconditions;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.TreeMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Utility for checking Alluxio environment.
+ */
+public final class Check {
+  private static final String USAGE = "USAGE: Check [task full name or prefix]\n\n"
+      + "Check environment for Alluxio";
+
+  private static final Options OPTIONS = new Options();
+
+  private static final Map<String, CheckTask> TASK_MAP = initializeTaskMap();
+
+  private static Map<String, CheckTask> initializeTaskMap() {
+    Map<String, CheckTask> taskMap = new TreeMap<>();
+    taskMap.put("master.port.available", new MasterRpcCheckTask());
```

Comment:
There are multiple ports including RPC, web, data. You can find the ports at http://www.alluxio.org/docs/master/en/Configuration-Settings.html